subsys/fs/zms/zms_priv.h

```diff
@@ -0,0 +1,50 @@
+/*  ZMS: Zephyr Memory Storage
+ *
+ * Copyright (c) 2024 BayLibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __ZMS_PRIV_H_
+#define __ZMS_PRIV_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * MASKS AND SHIFT FOR ADDRESSES
+ * an address in zms is an uint64_t where:
+ *   high 4 bytes represent the sector number
+ *   low 4 bytes represent the offset in a sector
+ */
+#define ADDR_SECT_MASK  GENMASK64(63, 32)
+#define ADDR_SECT_SHIFT 32
+#define ADDR_OFFS_MASK  GENMASK64(31, 0)
+#define SECTOR_NUM(x)   FIELD_GET(ADDR_SECT_MASK, x)
+
+#define ZMS_BLOCK_SIZE 32
+
+#define ZMS_LOOKUP_CACHE_NO_ADDR GENMASK64(63, 0)
+#define ZMS_HEAD_ID              GENMASK(31, 0)
+
+/* Allocation Table Entry */
+struct zms_ate {
+	uint32_t id;     /* data id */
+	uint32_t offset; /* data offset within sector */
+	uint16_t len;    /* data len within sector */
+	union {
+		uint32_t data_crc; /* crc for data */
+		uint32_t data;     /* used to store small size data */
+	};
+	uint8_t cycle_cnt; /* cycle counter for non erasable devices */
+	uint8_t crc8;      /* crc8 check of the entry */
+} __packed;
+
+BUILD_ASSERT(offsetof(struct zms_ate, crc8) == sizeof(struct zms_ate) - sizeof(uint8_t),
```

Comment:
Might be replaced by the following code to be ready to change the CRC field later.
```
BUILD_ASSERT(offsetof(struct zms_ate, crc8) ==
		 sizeof(struct zms_ate) - SIZEOF_FIELD(struct zms_ate, crc8),
		 "crc8 must be the last member");
```